synapse/handlers/auth.py

```diff
@@ -263,7 +293,7 @@ def _get_session_info(self, session_id):
         if not session_id:
             # create a new session
             while session_id is None or session_id in self.sessions:
-                session_id = stringutils.random_string(24)
+                session_id = stringutils.random_string_with_symbols(24)
```

Comment:
Its probably fine, but `random_string_with_symbols` will return a lot of silly symbols, so I've tended to avoid using them in public APIs (especially for anything that is used as query string params)